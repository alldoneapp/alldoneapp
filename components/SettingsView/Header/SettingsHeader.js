import React from 'react'
import { StyleSheet, View, Text } from 'react-native'
import Icon from '../../Icon'
import styles, { colors } from '../../styles/global'
import BackButton from './BackButton'
import { useSelector } from 'react-redux'
import { useTranslator, translate } from '../../../i18n/TranslationService'

export const SettingsHeader = ({}) => {
    useTranslator()
    const mobile = useSelector(state => state.isMiddleScreen)

    return (
        <View style={localStyles.container}>
            <View style={localStyles.upperHeader}>
                {mobile && (
                    <View style={localStyles.backButtonMobile}>
                        <BackButton />
                    </View>
                )}
                <View style={{ marginRight: 'auto', flex: 1 }}>
                    <Text style={[styles.title4, localStyles.title]}>{translate('Settings')}</Text>
                </View>
                <View style={[localStyles.icon]}>
                    <Icon name={'settings'} size={24} color={colors.Text03} />
                </View>
            </View>
        </View>
    )
}

const localStyles = StyleSheet.create({
    container: {
        height: 96,
        minHeight: 96,
        maxHeight: 96,
        flexDirection: 'column',
        justifyContent: 'space-between',
        marginBottom: 8,
    },
    upperHeader: {
        flexDirection: 'row',
        justifyContent: 'space-between',
    },
    title: {
        paddingTop: 32,
        height: 64,
        flex: 1,
    },
    icon: {
        marginTop: 36,
        flexDirection: 'row',
        alignItems: 'center',
    },
    backButtonMobile: {
        left: -16,
    },
})

export default SettingsHeader
