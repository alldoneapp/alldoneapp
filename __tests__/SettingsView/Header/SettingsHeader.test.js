/**
 * @jest-environment jsdom
 */

import React from 'react'
import SettingsHeader from '../../../components/SettingsView/Header/SettingsHeader'
import renderer from 'react-test-renderer'

describe('SettingsHeader component', () => {
    describe('SettingsHeader snapshot test', () => {
        it('should render correctly', () => {
            const tree = renderer.create(<SettingsHeader />).toJSON()
            expect(tree).toMatchSnapshot()
        })
    })
})
