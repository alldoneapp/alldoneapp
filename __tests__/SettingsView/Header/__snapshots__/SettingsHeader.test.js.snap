// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsHeader component SettingsHeader snapshot test should render correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "column",
      "height": 96,
      "justifyContent": "space-between",
      "marginBottom": 8,
      "maxHeight": 96,
      "minHeight": 96,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "marginRight": "auto",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#04142F",
              "fontFamily": "Roboto-Medium",
              "fontSize": 24,
              "lineHeight": 32,
            },
            Object {
              "flex": 1,
              "height": 64,
              "paddingTop": 32,
            },
          ]
        }
      >
        Settings
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "marginTop": 36,
          },
        ]
      }
    >
      <View>
        <Text
          style={
            Object {
              "color": "#8A94A6",
              "fontFamily": "alldone",
              "fontSize": 24,
              "opacity": 1,
            }
          }
        >
          î§­
        </Text>
      </View>
    </View>
  </View>
</View>
`;
